### PR TITLE
parser: Treat key as primary key in columndef

### PR DESCRIPTION
**Pull request by shenli**
Fix: https://github.com/pingcap/tidb/issues/961

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @zimulala @zxylvlp PTAL

**Comment by coocood** [comment]
I think we should change parser in batch. This little change has +3,002 −2,968 lines of code.

File: parser/parser.y [hunk 1]
```diff
@@ -864,13 +871,16 @@ ColumnOption:
...
-|	"PRIMARY" "KEY"
+|	PrimaryOpt "KEY" 
 	{
+		// KEY is normally a synonym for INDEX. The key attribute PRIMARY KEY 
+		// can also be specified as just KEY when given in a column definition. 
+		// See: http://dev.mysql.com/doc/refman/5.7/en/create-table.html
 		$$ = &ast.ColumnOption{Tp: ast.ColumnOptionPrimaryKey}
 	}
-|	"UNIQUE"
+|	"UNIQUE" %prec lowerThanKey
```
Comment (zxylvlp) [inline parser/parser.y]: Why does it necessary?

Comment (shenli) [inline parser/parser.y]: Or you will get a shift/reduce conflict.

Comment (zxylvlp) [inline parser/parser.y]: solve conflict unique and unique key?

Comment (shenli) [inline parser/parser.y]: Yes "unique key" can be treat as "unique key" or "unique"+"key", so we must specify its precedence.

**Comment by zxylvlp** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**